src/arithmetic/string_funcs/string_funcs.c

```diff
@@ -0,0 +1,325 @@
+/*
+* Copyright 2018-2019 Redis Labs Ltd. and Contributors
+*
+* This file is available under the Redis Labs Source Available License Agreement
+*/
+
+#include "string_funcs.h"
+#include "../func_desc.h"
+#include "../../util/arr.h"
+#include "../../util/rmalloc.h"
+#include <ctype.h>
+#include <assert.h>
+
+static void inline _toLower(const char *str, char *lower, short *lower_len) {
+	size_t str_len = strlen(str);
+	/* Avoid overflow. */
+	assert(*lower_len > str_len);
```

Comment:
Should this assertion be replaced with error reporting?

Reply:
The caller function already calculated strlen and set `lower_len`; there's a lot of unnecessary stuff here - I'll improve it.